common/src/main/java/io/pravega/common/concurrent/ExecutorServiceHelpers.java

```diff
@@ -56,6 +72,7 @@ public static ThreadFactory getThreadFactory(String groupName) {
             @Override
             public Thread newThread(Runnable r) {
                 Thread thread = new Thread(r, groupName + "-" + threadCount.incrementAndGet());
+                thread.setUncaughtExceptionHandler(new LogUncaughtExceptions());
```

Comment:
We could override `java.util.concurrent.ThreadPoolExecutor#afterExecute` of the `ScheduledExecutorService ` to log the exceptions. This is a guaranteed way of logging exceptions when using  `ThreadPoolExecutor`s.

Reply:
We could. I'm not sure it's better because it would be in the loop on every execution (and require subclassing) as opposed to just when an exception is thrown that is not being delivered to a future and would otherwise kill the thread.